cluster-endpoints/src/grpc_multiplex.rs

```diff
@@ -61,98 +63,75 @@ pub fn create_grpc_multiplex_blocks_subscription(
         info!("- connection to {}", grpc_source);
     }
 
+    // let confirmed_blocks_stream = create_confirmed_blocks_stream(&grpc_sources);
+
+    // let finalized_blockmeta_stream = create_finalized_blockmeta_stream(&grpc_sources);
+
     // return value is the broadcast receiver
     let (producedblock_sender, blocks_output_stream) =
         tokio::sync::broadcast::channel::<ProducedBlock>(1000);
 
     let jh_block_emitter_task = {
         tokio::task::spawn(async move {
-            loop {
-                let confirmed_blocks_stream = {
-                    let commitment_config = CommitmentConfig::confirmed();
-
-                    let mut streams = Vec::new();
-                    for grpc_source in &grpc_sources {
-                        let stream = create_geyser_reconnecting_stream(
-                            grpc_source.clone(),
-                            GeyserFilter(commitment_config).blocks_and_txs(),
-                        );
-                        streams.push(stream);
-                    }
+            // by blockhash
+            let mut recent_confirmed_blocks = HashMap::<String, ProducedBlock>::new();
+            let mut confirmed_blocks_stream = Box::pin(create_confirmed_blocks_stream(&grpc_sources));
+            let mut finalized_blockmeta_stream = Box::pin(create_finalized_blockmeta_stream(&grpc_sources));
 
-                    create_multiplexed_stream(streams, BlockExtractor(commitment_config))
-                };
-
-                let finalized_blockmeta_stream = {
-                    let commitment_config = CommitmentConfig::finalized();
+            let sender = producedblock_sender;
+            let mut delay_reconnect_until = Instant::now() + STREAM_RECONNECT_COOLDOWN;
+            let mut block_heartbeat_tick = tokio::time::interval(Duration::from_secs(5));
+            let mut cleanup_tick = tokio::time::interval(Duration::from_secs(5));
+            let mut last_finalized_slot: Slot = 0;
+            loop {
+                tokio::select! {
+                    confirmed_block = confirmed_blocks_stream.next() => {
```

Comment:
I feel the streams are not working well, the performance issues that I found was related to stream when I moved back to channel it is not working very well. May be thing that you can check on your side.